app/models/user_profile.rb

```diff
@@ -0,0 +1,12 @@
+# typed: strict
+class UserProfile < ApplicationRecord
+  belongs_to :user
+
+  # Define the attributes with their respective data types and constraints
+  attribute :id, :integer
+  attribute :created_at, :datetime
+  attribute :updated_at, :datetime
+  attribute :order_history, :text
+  attribute :settings, :text
+  attribute :user_id, :integer
```

Comment:
Similar to the Customization model, the UserProfile model defines attributes that are already managed by the migrations. This redundancy can be removed to avoid potential confusion and maintain consistency with the database schema.